library/aes.c

```diff
@@ -55,8 +55,15 @@
 #if !defined(MBEDTLS_AES_ALT)
 
 /* Implementation that should never be optimized out by the compiler */
-static void mbedtls_zeroize( void *v, size_t n ) {
-    volatile unsigned char *p = (unsigned char*)v; while( n-- ) *p++ = 0;
+static void mbedtls_zeroize( void *v, size_t n )
+{
+    if( n > 0 )
```

Comment:
Unlike the other implementations of zeroize that call `memset`, we probably don't need this `if` here as there is no undefined behavior. If n is 0, the loop is not executed. The local volatile pointer assignment won't do anything meaningful, since we don't use `p` at all when n is 0.

Reply:
There's no bug to fix in 2.7, is there?